### ISSUE TITLE
Option to check colour contrast 

### ISSUE THREAD

**Issue by KaneCheshire** [KaneCheshire/PixelTest#30]
It would be really ace if it's feasible to somehow fail a test if the view fails colour contrast accessibility standards

**Comment by KaneCheshire** [comment]
This won't be quick or super easy to implement but would be nice to crack. Some things to consider:

- Contrast checks shouldn't really fail for backgrounds over other backgrounds, only text. Might mean doing something like finding all the labels in a view and snapshotting the label's frame (including whatever is behind it), and checking that contrast ratio
- Transparent backgrounds will throw off the test, so might be a good idea to add a white background for testing contrast by default
- This should be opt-in for all tests and on a per-view basis

**Comment by Neil3079** [comment]
I wonder if we need actual snapshotting logic for this, maybe it can be done by injecting the colour of the labels font and background colours of layers, though might cause issues when images are backgrounds...

**Comment by KaneCheshire** [comment]
Definitely an option, but my gut is telling me to go with snapshotting the frame of each label within the "main" view, if that makes sense?
